### PR TITLE
feat(route/yahoo/news): add HK-exclusive `listId` mode

### PR DESCRIPTION
**Pull request by tpnonthealps**
## Involved Issue / 该 PR 相关 Issue

N/A

## Example for the Proposed Route(s) / 路由地址示例

```routes
/yahoo/news/list/hk/5a8a0aa0-0ab3-11e8-b3dc-d990c79d6cb1
/yahoo/news/list/hk/1cc44280-facb-11e9-ad7c-f3ba971275c8
/yahoo/news/list/hk/40023670-facc-11e9-9dde-9175ff306602
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
```
as some providers show no contents in archive …
```

### CODE REVIEW

File: lib/routes/yahoo/news/listid.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    | 最新新聞     | \`09fcf7b0-0ab2-11e8-bf1f-4d52d4f79454\` |
+    | 東方日報     | \`33ddd580-0ab3-11e8-bfe1-4b555fb1e429\` |
+    | now.com      | \`01b4d760-0ab4-11e8-af3a-54037d3dced3\` |
+    | am730        | \`c4842090-0ab2-11e8-af7f-041a72ce7398\` |
+    | BBC          | \`4d3fc9a0-fac8-11e9-87f2-564ca250983e\` |
+    | 信報財經新聞 | \`5a8a0aa0-0ab3-11e8-b3dc-d990c79d6cb1\` |
+    | 香港電台     | \`b4bfc2d0-0ab3-11e8-bf9f-c888fc09923f\` |
+    | 法新社       | \`1cc44280-facb-11e9-ad7c-f3ba971275c8\` |
+    | Bloomberg    | \`40023670-facc-11e9-9dde-9175ff306602\` |
+    | 香港動物報   | \`6058fa9c-d74d-487a-8b49-aa99a2a2978e\` |`,
```
Comment (TonyRL) [inline lib/routes/yahoo/news/listid.ts:39]: All providers can be subscribed using `/yahoo/news/provider/:region/:providerId`

`東方日報`: `/yahoo/news/provider/hk/orientaldaily_hk_156`
`now.com`: `/yahoo/news/provider/hk/now新聞網`
`am730`: `/yahoo/news/provider/hk/am730_578`
`BBC`: `/yahoo/news/provider/hk/bbc_trad_chinese_998`
`信報財經新聞`: `/yahoo/news/provider/hk/hkej.com`
`香港電台`: `/yahoo/news/provider/hk/rthk.hk.local`
`法新社`: `/yahoo/news/provider/hk/afp.com.hk`
`Bloomberg`: `/yahoo/news/provider/hk/bloomberg_chinese_traditional_757`
`香港動物報`: `/yahoo/news/provider/hk/hk_animal_post_231`

File: lib/routes/yahoo/news/listid.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    // console.log('Response:', response.stream_items);
+    // console.log('Type of response:', typeof response.stream_items);
+    // console.log('Is response an array?', Array.isArray(response.stream_items));
+    const list = parseList(region, response.stream_items);
+
+    const items = await Promise.all(list.map((item) => parseItem(item, cache.tryGet)));
+
+    return {
+        title: `Yahoo 新聞`,
```
Comment (TonyRL) [inline lib/routes/yahoo/news/listid.ts:58]: Consider include the provider name in RSS title.

File: lib/routes/yahoo/news/listid.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    | 信報財經新聞 | \`5a8a0aa0-0ab3-11e8-b3dc-d990c79d6cb1\` |
+    | 香港電台     | \`b4bfc2d0-0ab3-11e8-bf9f-c888fc09923f\` |
+    | 法新社       | \`1cc44280-facb-11e9-ad7c-f3ba971275c8\` |
+    | Bloomberg    | \`40023670-facc-11e9-9dde-9175ff306602\` |
+    | 香港動物報   | \`6058fa9c-d74d-487a-8b49-aa99a2a2978e\` |`,
+};
+
+async function handler(ctx) {
+    const { region, listId } = ctx.req.param();
+    if (!['hk'].includes(region)) {
```
Comment (TonyRL) [inline lib/routes/yahoo/news/listid.ts:44]: LISTID also works for `tw` region. For example, `4fc44c36-20be-4f36-9ad5-5b4eff8796a9`

File: lib/routes/yahoo/news/listid.ts (hunk 1 above)
Comment (TonyRL) [inline lib/routes/yahoo/news/listid.ts:39]: Indent with 2 or 0 spaces

File: lib/routes/yahoo/news/listid.ts (hunk 3 above)
Comment (tpnonthealps) [reply lib/routes/yahoo/news/listid.ts:44]: OK, but I cant provide a chart for routes in `tw`

File: lib/routes/yahoo/news/listid.ts (hunk 2 above)
Comment (tpnonthealps) [reply lib/routes/yahoo/news/listid.ts:58]: any suggestions to include it without hardcode names in the code?

Comment (TonyRL) [reply lib/routes/yahoo/news/listid.ts:58]: items[0].author

**Pull request merged**